src/layers.rs

```diff
@@ -57,6 +57,15 @@ pub struct Layer {
     /// The ID of the layer, as shown in the editor.
     /// Layer ID stays the same even if layers are reordered or modified in the editor.
     pub id: u32,
+
+    /// Derived from the first chunk. There's no good way to generally find out
+    /// the chunk size, even though
+    /// [editorsettings doc](https://doc.mapeditor.org/en/stable/reference/tmx-map-format/#chunksize)
+    /// suggests it's the same for the entire map.
+    /// Maybe in future Tiled will support different chunk sizes in one map, but
+    /// I doubt it.
+    pub chunk_width: u32,
+    pub chunk_height: u32,
```

Comment:
At the moment Tiled only supports a fixed chunk size, where the size is actually stored on the map, but as you noticed this is an editor setting. I could imagine to add support for dynamic chunk sizes in the future, but it would likely be only added on demand. While that might be unlikely, I'm not so keen on adding these members.

Reply:
Currently, in order to retrieve a chunk, the code needs to know its top-left corner. E.g. in order to retrieve tile at (17, 17) from `Infinite(HashMap<(i32, i32), ...)`, code needs to deduce the (16,16) key; trying (17,17) will not find it.

This is why I kinda needed these fields.

Ideal general solution would be to use a quad-tree to map coordinates to chunks. But as the values don't really change, I thought a simpler solution would work fine for many years. Quad-tree also has a logarithmic access time, and direct indexing is constant.

Another solution is to retrieve a (0,0) chunk on each `get_tile(x, y)` call, read ITS size, assume that all chunks have this size, and then evaluate index of the needed chunk. But it's still one extra operation where I don't want it, namely in a rendering code.

Here are the options I see. What do you say?

Reply:
I'd prefer the hashmap to a quad tree, indeed because lookups should be very fast. Alternatively the hashmap could even be an array of chunk references, which could waste some memory but avoids the hashing. The second solution is definitely also something we'd want to avoid.

But, could you share your thoughts on the suggestion to use a fixed chunk size in `rs-tiled`, independent from the chunks found in the TMX file, and while loading the chunks in the TMX file, to just copy over the tile data in the fixed chunks data structure? This is the way Tiled works internally.